detectors/node/opentelemetry-resource-detector-docker/src/detectors/DockerDetector.ts

```diff
@@ -51,21 +53,39 @@ export class DockerCGroupV1Detector implements Detector {
 
   private async _getContainerId(): Promise<string | undefined> {
     try {
-      const rawData = await DockerCGroupV1Detector.readFileAsync(
+      let rawData = await DockerDetector.readFileAsync(
         this.DEFAULT_CGROUP_PATH,
         this.UTF8_UNICODE
       );
-      const splitData = rawData.trim().split('\n');
+      let splitData = rawData.trim().split('\n');
       for (const str of splitData) {
         if (str.length >= this.CONTAINER_ID_LENGTH) {
           return str.substring(str.length - this.CONTAINER_ID_LENGTH);
         }
       }
+
+      // If this code is reached - we then check for the V2_PATH
+      rawData = await DockerDetector.readFileAsync(
+        this.DEFAULT_CGROUP_V2_PATH,
+        this.UTF8_UNICODE
+      );
+
+      splitData = rawData.trim().split('\n');
+      for (let str of splitData) {
+        if (str.includes(this.HOSTNAME)) {
+          str = str
+            .split('/')
+            .filter(s => s.length >= this.CONTAINER_ID_LENGTH)[0];
+          if (str.length >= this.CONTAINER_ID_LENGTH) {
+            return str.substring(str.length - this.CONTAINER_ID_LENGTH);
+          }
+        }
+      }
     } catch (e) {
```

Comment:
Question: When V1 tries to `readFileAsync(this.DEFAULT_CGROUP_PATH, ..)`, if the file is missing I think it will throw "Error: ENOENT: no such file or directory, open '/proc/self/cgroup'" (at least on my setup).

In this case, it will fall into the `catch` and will not attempt to try with DEFAULT_CGROUP_V2_PATH. Am I missing something?

Reply:
The assumption is and I have confirmed that the file would exist - it would just not have the id in it. and then it will fall back to the v2 implementation.

Reply:
Great, thank you for validating.
Can I please ask you to add this insight as comment, so future developers looking at this code will know that it's intended, and confirmed to behave in the right way?